### PR TITLE
Running indicator

### PR DESCRIPTION
**Pull request by aaron-binary**

### CODE REVIEW

File: src/botPage/view/blockly/index.js [hunk 1]
```diff
@@ -440,7 +440,17 @@ while(true) {
...
+            const elStopButtons = document.querySelectorAll('#stopButton, #summaryStopButton');
+
+            elRunButtons.forEach(el => {
+                const elRunButton = el;
+                elRunButton.style.display = 'initial';
+            });
+            elStopButtons.forEach(el => {
+                const elStopButton = el;
+                elStopButton.style.display = null;
+            });
```
Comment (MahboobehMohammadi) [inline src/botPage/view/blockly/index.js]: @aaron-binary you copy this code almost 4 times. can it all goes in some "setBotButtonsVisibility` or st like that.

File: src/botPage/bot/Interpreter.js [hunk 2]
```diff
@@ -149,8 +149,10 @@ export default class Interpreter {
     }
     terminateSession() {
         this.$scope.api.disconnect();
-        globalObserver.emit('bot.stop');
         this.stopped = true;
+
+        globalObserver.emit('bot.stop');
```
Comment (MahboobehMohammadi) [inline src/botPage/bot/Interpreter.js]: @aaron-binary why can we just change the buttons state when we emit bot.stop, why we need to keep it in a state?

File: src/botPage/view/View.js [hunk 3]
```diff
@@ -627,31 +635,54 @@ export default class View {
...
+            });
+
             if (error.error && error.error.error.code === 'InvalidToken') {
                 removeAllTokens();
                 updateTokenList();
                 this.stop();
             }
         });
 
+        globalObserver.register('bot.running', () => {
```
Comment (MahboobehMohammadi) [inline src/botPage/view/View.js]: @aaron-binary does this `globalObserver` have something like `reaction` so it listen to state changes and when ever bot state change to running or not running , disable and enable the `buttons` once only.

File: src/botPage/view/blockly/index.js (hunk 1 above)
Comment (aaron-binary) [reply src/botPage/view/blockly/index.js]: Each of the blocks does different things, e.g. in one case the block will just be hidden, in another, the block will be hidden and disabled, and in the other, the block is just disabled.

File: src/botPage/view/View.js (hunk 3 above)
Comment (aaron-binary) [reply src/botPage/view/View.js]: Through the observer you can `observer.register('name')` to a named event, once that event is emitted (through `observer.emit('name')`) the callback will be executed. In some cases, the buttons should not be enabled when the bot is not running, e.g. when the user is fetching `contracts_for` from the API.

**Pull request merged**